beps/0001-notifications-system/README.md

```diff
@@ -380,10 +380,39 @@ interface SignalApi {
 - Render dynamic values with various different React elements such as the `EntityRefLink` for entity references (for example `{{ user:default/john.doe }}`) in the notification payload
 - Handle `link` values that use route references. For example instead hard-coding link to `/catalog/default/component/artist-web` it should be possible to use `catalogPlugin.catalogEntity` route reference as a link of the notification. This should also allow using parameters to be passed to the route reference. Links to external systems are still supported.
 - Add support for `analyticsApi` to notification actions like marking notifications done, saved or opening links in the notifications
-- Add support for user settings to control how notifications are shown to the user and which notifications user wants to receive. This should also include support for different `NotificationProcessor`s that can send notification to external systems
 - Add a sound to be played when notification is received
 - Add i18n internationalization support for the notification payload
 
+### User specific notification settings
+
+To allow the users more fine-grained control of notifications, user must have an option to unsubscribe from specific origins, topics and channels.
+
+By default, all notifications are enabled for the user in all notification channels. Frontend plugin provides an user interface to change these settings for example as described in the following image:
+
+![user notification settings UI example](./UserNotificationSettings.png)
+
+Each notification processor with `send` functionality will be listed as a separate channel among the Backstage internal notification channel `Web`. For this, each notification processor must implement a function to get a human readable name to be rendered in the frontend.
+
+User must be able to save these settings to the notification database, and they are be checked each time a new notification is sent to the user.
+
+Disabling a notification from specific origin or topic in the `Web` channel will not remove the old notifications being visible in the frontend from that source but instead, it only prevents new notifications being sent.
+
+For performance reasons the notification settings will ignore all broadcast notifications and those will be sent to all users despite their origin or channel.
+
+```ts
+export type NotificationSetting = {
+  origin: string;
+  topic?: string;
+  channels: Record<string, boolean>;
+};
+
+export type UserNotificationSettings = {
+  settings: NotificationSetting[];
```

Comment:
What about
```suggestion
  routes: NotificationSetting[];
```
Or another variant on `userNotificaionRoutes`?

There can be additional settings needed in the future, i.e. preselection of filters.

Reply:
Don't understand, could you elaborate?

Reply:
I am pointing to a way how we persist/transfer all settings related to the Notifications in an extensible way.

The so far proposed settings are in fact routing - whether a processor will handle the particular message.

In the (near-) future, we can go beyond that settings. I.e. we can add a way how the user can persist his selection of filters on the NotificationsPage and so create "a personalized view". Or something else. The `UserNotificationSettings` is a nice wrapper for all these user-specific settings. 

In other words, the proposed `UserNotificationSettings` has the only property called `settings`. Any future additions will be kind of `settings` as well, just for something else.